### PR TITLE
Add some missing methods for servers

### PR DESCRIPTION
**Pull request by igornatanael**
This patch adds support for Servers: Hardware, Profile, Templates and Connections
as well as its unit tests

### CODE REVIEW

File: examples/scripts/get-connections.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+def getconn(srv):
+    ret = srv.get_connections()
+    pprint(ret)
+
+
+def main():
+    parser = argparse.ArgumentParser(add_help=True,
+                        formatter_class=argparse.RawTextHelpFormatter,
+                                     description='''
+    Display Server Resources
```
Comment (danielreed) [inline examples/scripts/get-connections.py]: should this be 'Display connection information' or something similar?

File: examples/scripts/get-envconf.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                        default='Administrator',
+                        help='''
+    HP OneView Username''')
+    parser.add_argument('-p', dest='passwd', required=True,
+                        help='''
+    HP OneView Password''')
+    parser.add_argument('-n', dest='name',
+                        required=False,
+                        help='''
+    The name of the server hardware resource to be returned its utilization.
```
Comment (danielreed) [inline examples/scripts/get-envconf.py]: should line 93 describe that the script is returning environmental configuration

File: examples/scripts/get-envconf.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+        else:
+            ret = 'This server does not exist.'
+    pprint(ret)
+
+
+def main():
+    parser = argparse.ArgumentParser(add_help=True,
+                        formatter_class=argparse.RawTextHelpFormatter,
+                                     description='''
+    Display Server Resources
```
Comment (danielreed) [inline examples/scripts/get-envconf.py]: maybe this should be Display Server Environmental Configuration

File: examples/scripts/get-serverbios.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+        else:
+            ret = 'This server does not exist.'
+    pprint(ret)
+
+
+def main():
+    parser = argparse.ArgumentParser(add_help=True,
+                        formatter_class=argparse.RawTextHelpFormatter,
+                                     description='''
+    Display Server Resources
```
Comment (danielreed) [inline examples/scripts/get-serverbios.py]: should this be updated to show that server bios settings will be returned

Comment (igornatanael) [inline examples/scripts/get-serverbios.py]: I really didn't pay attention when I was reusing the code.
Thanks, I'm already fixing all this.

**Pull request merged**